executor.go

```diff
@@ -278,15 +279,26 @@ func executeFields(p ExecuteFieldsParams) *Result {
 		p.Fields = map[string][]*ast.Field{}
 	}
 
+	var wg sync.WaitGroup
+	wg.Add(len(p.Fields))
+
+	var resultsMutex sync.Mutex
 	finalResults := map[string]interface{}{}
 	for responseName, fieldASTs := range p.Fields {
-		resolved, state := resolveField(p.ExecutionContext, p.ParentType, p.Source, fieldASTs)
-		if state.hasNoFieldDefs {
-			continue
-		}
-		finalResults[responseName] = resolved
+		go func(responseName string, fieldASTs []*ast.Field) {
```

Comment:
A half year ego I wanted to make similar changes, but then decided that spawning a goroutine on every resolve might be much more expensive than doing it in series. So now I wonder, have you checked the impact on the execution time and resource allocation? Does it really make sense?